frontend/src/components/custom-tools/export-tool/ExportTool.jsx

```diff
@@ -0,0 +1,228 @@
+                  <Popconfirm
+                    key={`${item.id}-delete`}
+                    title="Delete the User"
+                    description={`Are you sure to remove ${item?.email}?`}
+                    okText="Yes"
+                    cancelText="No"
+                    icon={<QuestionCircleOutlined />}
+                    onConfirm={(event) => handleDeleteUser(item?.id)}
+                  >
+                    <Typography.Text>
+                      <DeleteOutlined className="action-icon-buttons" />
+                    </Typography.Text>
+                  </Popconfirm>
+                </div>
+              }
+            >
+              <List.Item.Meta
+                title={
+                  <>
+                    <Avatar
+                      className="shared-user-avatar"
+                      icon={<UserOutlined />}
+                    />
+                    <Typography.Text className="export-username">
+                      {item.email}
+                    </Typography.Text>
+                  </>
+                }
+              />
+            </List.Item>
+          )}
+        />
+        <Typography>
+          Exported Tools are always shared with the Prompt Studio projects owner
+          and users in addition to other users it is explicitly shared with
+        </Typography>
+      </>
+    );
+  }
+
+  return (
+    toolDetails && (
+      <Modal
+        title={"Export settings"}
+        open={open}
+        onCancel={() => setOpen(false)}
+        maskClosable={false}
+        centered
+        closable={true}
+        okText={"Apply"}
+        onOk={() =>
+          onApply(selectedUsers, toolDetails, sharingOption === SHARE_ALL)
+        }
+        className="share-permission-modal"
+      >
+        {loading ? (
+          <SpinnerLoader />
+        ) : (
+          <>
+            {allUsers.length > 1 && (
+              <Radio.Group
+                onChange={onChange}
+                value={sharingOption}
+                className="export-per-radio"
+              >
+                <Radio value={SHARE_ALL}>Share with everyone</Radio>
+                <Radio value={SHARE_CUSTOM}>Custom share</Radio>
+              </Radio.Group>
+            )}
+            {sharingOption !== SHARE_ALL && (
+              <Select
+                filterOption={filterOption}
+                showSearch
+                size={"middle"}
+                placeholder="Search"
+                value={null}
+                className="export-permission-search"
+                onChange={(selectedUser) => {
+                  const isUserSelected = selectedUsers.includes(selectedUser);
+                  if (!isUserSelected) {
+                    setSelectedUsers([...selectedUsers, selectedUser]);
+                  }
+                }}
+                options={filteredUserList.map((user) => ({
+                  label: user.email,
+                  value: user.id,
```

Comment:
@johnyrahul Please use [optional chaining](https://www.geeksforgeeks.org/javascript-optional-chaining/) here (and other places, if any) to safely access properties.